python/lsst/qa/explorer/classifyObjects.py

```diff
@@ -0,0 +1,495 @@
+
+        Notes
+        -----
+        The columns added are called `magPsf + filter name`, `magPsfErr + filter name`,
+        `magModel + filter name` and `magModelErr + filter name`. They are added for every filter given in the
+        filters argument.
+        """
+
+        colsIn = [self.config.psfColName, self.config.modelColName]
+        colsOut = ["magPsf", "magModel"]
+
+        for band in filters:
+
+            calib = dataRef.get("deepCoadd_calexp_calib", filter=band)
+
+            for (i, col) in enumerate(colsIn):
+
+                goodFlux = np.where((np.isfinite(cat[band][col])) & (cat[band][col] > 0.0))[0]
+
+                magCol = np.array([np.nan]*len(cat))
+                magErrCol = np.array([np.nan]*len(cat))
+
+                flux = cat[band][col].iloc[goodFlux]
+                fluxErr = cat[band][col + "Sigma"].iloc[goodFlux]
+
+                mags, magErrs = calib.getMagnitude(flux.values, fluxErr.values)
+
+                magCol[goodFlux] = mags
+                magErrCol[goodFlux] = magErrs
+
+                cat["featuresSGSep", colsOut[i] + band] = magCol
+                cat["featuresSGSep", colsOut[i] + "Err" + band] = magErrCol
+
+        return cat
+
+    def addSeeing(self, cat, filters):
+        """Add seeing defined by seeing = (xx*yy - xy*xy)**0.25 where xx, yy and xy are enteries in the
+        base_SdssShape_psf matrix
+
+        Parameters
+        ----------
+        cat : `pandas.core.frame.DataFrame`
+        filters : `list`
+
+        Returns
+        -------
+        cat : `pandas.core.frame.DataFrame`
+            Data frame with new index added to store columns used by the classifiers
+
+        Notes
+        -----
+        The new columns added are called `seeing + filter name` and are added for every filter in the list of
+        filters provided.
+        """
+
+        seeingCol = self.config.seeingColName
+
+        for band in filters:
+            xx = cat[band][seeingCol + "_xx"]
+            xy = cat[band][seeingCol + "_xy"]
+            yy = cat[band][seeingCol + "_yy"]
+
+            seeing = (xx*yy - xy*xy)**0.25
+
+            cat["featuresSGSep", "seeing" + band] = seeing
+
+        return cat
+
+    def addColors(self, cat, filters):
+        """Add all available colors
+
+        Parameters
+        ----------
+        cat : `pandas.core.frame.DataFrame`
+        filters : `list`
+
+        Returns
+        -------
+        cat : `pandas.core.frame.DataFrame`
+            Data frame with new index added to store columns used by the classifiers
+
+        Notes
+        -----
+        len(filters)-1 new columns are added giving each filter minus the filter after it when the filters
+        are arranged in wavelength order. These columns are named `filter[n]-filter[n+1]`. E.g. for HSC:
+        `HSC-G-HSC-R`, `HSC-R-HSC-I`, `HSC-I-HSC-Z` and `HSC-Z-HSC-Y`. Needs to be run after addMagnitudes so
+        that the magnitude columns are already present.
```

Comment:
This is great!

Two minor suggestions (which also apply to the `addMB` method below:
* Have "Needs to be run..." a new paragraph (after a blank line) since it is unrelated to filter names and is important enough that it needs to stand out.
* What happens if the magnitude columns are missing? I'm hoping it raises an exception, in which case please document that in a "Raises" section.

Reply:
I've put in a bit checking the existence of the required columns in runDataRef, the same time as it checks that the required filters are there.